src/TreeView/TreeView.js

```diff
@@ -0,0 +1,353 @@
+                React.cloneElement(child, {
+                    expandData,
+                    onExpandClick,
+                    isExpanded,
+                    // Increment child list level
+                    level: level + 1
+                }) :
+                null;
+        });
+
+        // Render child TreeCols
+        const cells = React.Children.map(children, (child, index) => {
+            const isTreeCol = child.type && child.type.name === 'TreeCol';
+            const isFirstTreeCol = index === 0 && isTreeCol;
+
+            // Add control class to first TreeCol element
+            const className = classnames({
+                'fd-tree__col--control': isFirstTreeCol
+            });
+
+            // Add expand button to first TableCell if parent list
+            const newChildren = isFirstTreeCol && childList[0] ? (
+                <div>
+                    <button
+                        aria-controls={this.rowId}
+                        aria-label={isExpanded ? 'collapse' : 'expand'}
+                        aria-pressed={isExpanded}
+                        className='fd-tree__control'
+                        onClick={() => onExpandClick(this.rowId)} />
+                    {child.props && child.props.children}
+                </div>
+            ) : child.props && child.props.children;
+
+            return isTreeCol ?
+                React.cloneElement(child, { className, children: newChildren }) :
+                null;
+        });
+
+        return (
+            <li
+                {...rest}
+                aria-expanded={isExpanded}
+                className='fd-tree__item'
+                id={this.rowId}
+                role='treeitem'>
+                <div className='fd-tree__row'>
+                    {cells}
+                </div>
+                {childList}
+            </li>
+        );
+    }
+}
+
+TreeRow.propTypes = {
+    children: PropTypes.node,
+    expandData: PropTypes.object,
+    level: PropTypes.number,
+    onExpandClick: PropTypes.func
+};
+
+TreeRow.defaultProps = {
+    expandData: {},
+    level: 0,
+    onExpandClick: () => {}
+};
+
+TreeRow.propDescriptions = {
+    expandData: '_INTERNAL USE ONLY._',
+    level: '_INTERNAL USE ONLY._',
+    onExpandClick: '_INTERNAL USE ONLY._'
+};
+
+export class Tree extends Component {
+    render() {
+        const {
+            children,
+            expandData,
+            level,
+            isExpanded,
+            onExpandClick,
+            ...rest
+        } = this.props;
+
+        const className = classnames({
+            'fd-tree': level === 0,
+            'fd-tree__group': level > 0,
+            [`fd-tree__group--sublevel-${level}`]: level > 0,
+            'is-hidden': !isExpanded
+        });
+
+        return (
+            <ul
+                {...rest}
+                aria-hidden={!isExpanded}
+                className={className}
+                role={level === 0 ? 'tree' : 'group'}>
```

Comment:
i see this role for the tree. is there some sort of role relationship for the `<treehead>` with the tree? @jeffredodd?

Reply:
After looking at the WAI again, there is no concept of a header (`TreeHead`) for a `TreeView`.  However, there is a Tree Grid spec that defines a role of `treegrid` on a `<table>` element.  It appears to require standard table markup and uses some additional roles and aria attributes as well.  It really seems like this should be a separate component, but would need to be spec'd at the Fiori Fundamentals first.

See https://www.w3.org/TR/wai-aria-practices-1.1/examples/treegrid/treegrid-1.html